[UI] CC-4031: change from Action, a and button to hds::Button

valeriia-ruban opened the pull request:
### Description
Changes UI elements as `<a />`, `<button />` and `<Action />` to `<Hds::Button>`
- some outdated button styles are kept for `dropdown` current logic. Updating dropdown to Hds:Dropdown will be covered in another ticket ([CC-4315](https://hashicorp.atlassian.net/browse/CC-4315))
- partition page crashing is fixed 

<img width="1181" alt="image" src="https://user-images.githubusercontent.com/10027860/219207010-6d90eeb8-8cfc-47d1-a9d6-b6a682fe7d4f.png">
<img width="1171" alt="image" src="https://user-images.githubusercontent.com/10027860/219207073-f86cc5fb-41f7-4534-b494-c9cac0a2fbbd.png">
<img width="707" alt="image" src="https://user-images.githubusercontent.com/10027860/219207141-7308bc6d-ef26-41b8-a268-dfc972bebef0.png">
<img width="902" alt="image" src="https://user-images.githubusercontent.com/10027860/219207209-f9dfd482-f066-4f91-bc95-d5e727ab9a0b.png">
<img width="1177" alt="image" src="https://user-images.githubusercontent.com/10027860/219207355-52f2a574-4ae2-47c2-842c-c93be621f58c.png">

### Testing & Reproduction steps
Enable feature flags for `acrs`, `partitions`, `namespaces`, `peers`
Check button and link elements visually

### Links
https://hashicorp.atlassian.net/browse/CC-4031
https://hashicorp.atlassian.net/browse/CC-4329 - fixed as a part of updates

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern


[CC-4315]: https://hashicorp.atlassian.net/browse/CC-4315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jherschman reviewed (changes requested):
The changes are looking great ✨! I noted a few instances where we are using a button class instead of a standalone link class. Best practice for any links that leave the app should use a standalone link style instead of a button, I should have called that out in the figma file! 

I also noticed that there is a difference in font weight for the filters on each page (looks like it's a thinner weight). And I noticed on the Peers detail page, within the "exported services" and "server address" tabs the border / stroke is thicker and has a left and right border where it didn't used to. I couldn't find what style is impacting either of these changes (see attached screenshots).  


<img width="1566" alt="Screenshot 2023-02-16 at 11 22 37 AM" src="https://user-images.githubusercontent.com/91899352/219426394-3731d07e-153e-464a-bb53-9f845f119e76.png">
<img width="1597" alt="Screenshot 2023-02-16 at 11 22 45 AM" src="https://user-images.githubusercontent.com/91899352/219426395-727e0d85-4d71-49eb-b661-97b449b4801a.png">
<img width="1601" alt="Screenshot 2023-02-16 at 10 59 41 AM" src="https://user-images.githubusercontent.com/91899352/219426397-f5041fad-854b-4e34-b4f7-faeba3db1df4.png">
<img width="1609" alt="Screenshot 2023-02-16 at 11 00 11 AM" src="https://user-images.githubusercontent.com/91899352/219426400-3109a268-766a-4c85-a8d2-c8c557446f23.png">

WenInCode commented on ui/packages/consul-ui/app/components/consul/intention/notice/custom-resource/index.hbs:
```diff
@@ -15,7 +15,14 @@ as |notice|>
-      <a href="{{env 'CONSUL_DOCS_URL'}}/k8s/crds" target="_blank" rel="noopener noreferrer">Learn more about CRDs</a>
+      <Hds::Button
+        @href="{{env 'CONSUL_DOCS_URL'}}/k8s/crds"
+        @text='Learn more about CRDs'
```
Yes, it seems to be  that way according to the `Alert` docs in [Helios](https://helios.hashicorp.design/components/alert?tab=code).

valeriia-ruban replied:
Updated here 
<img width="1369" alt="image" src="https://user-images.githubusercontent.com/10027860/219811567-f6c4c7bd-cb62-45af-8500-0d34cb3c9c98.png">

but for sure, it will be a part of `Alert` update to `Hds`

WenInCode commented on ui/packages/consul-ui/app/templates/dc/acls/policies/-form.hbs:
```diff
@@ -28,53 +28,86 @@
+          <Hds::Button
+            @text='Save'
+            type='submit'
+            {{ action 'create' item}}
```
I think you'll have to do the same thing we talked about for roles here. It's also using the same mixin.

valeriia-ruban replied:
updated where found disabled + {{action}} on a button

WenInCode commented on ui/packages/consul-ui/app/components/role-selector/index.hbs:
```diff
@@ -56,27 +62,40 @@ as |modal|>
+          type='submit'
+          @text='Create and apply'
+          @isLoading={{policy.isSaving}}
+          {{action 'dispatch' 'save' (array policy item.Policies (action (mut state) 'role'))}}
```
Interesting that this doesn't have the same issues as the other disabled button

valeriia-ruban commented on ui/packages/consul-ui/translations/routes/en-us.yaml:
```diff
@@ -5,8 +5,8 @@ dc:
-        link: |
-          <a href="{CONSUL_DOCS_URL}/architecture/improving-consul-resilience#strategies-to-increase-fault-tolerance" target="_blank" rel="noopener noreferrer">Learn how to improve fault tolerance</a>
+        link-text: Learn how to improve fault tolerance
+        link: '/architecture/improving-consul-resilience#strategies-to-increase-fault-tolerance'
```
Nice catch! 

I've changed to Standalone link and fixed href (see on screenshot below)
<img width="1270" alt="image" src="https://user-images.githubusercontent.com/10027860/219810037-3afd4562-a4ca-403b-8245-4a83addcb870.png">

Let me know if we still need to change something. thank you for the VQA, it is amazing to have 👀 on these changes

valeriia-ruban commented on ui/packages/consul-ui/translations/routes/en-us.yaml:
```diff
@@ -34,25 +34,16 @@ dc:
+        links:
+          license-expiration:
+            text: License expiration
+            link: '/enterprise/license/faq#q-is-there-a-grace-period-when-licenses-expire'
```
fixed to standalone link and fixed hrefs for every link in License page
<img width="801" alt="image" src="https://user-images.githubusercontent.com/10027860/219812509-a8cee836-0248-4a38-b170-a6de6945186b.png">

valeriia-ruban commented on ui/packages/consul-ui/app/components/consul/acl/disabled/index.hbs:
```diff
@@ -15,11 +15,25 @@
-          <a href="{{env 'CONSUL_DOCS_LEARN_URL'}}/consul/security-networking/production-acls" rel="noopener noreferrer" target="_blank">Follow the guide</a>
+        <li>
+          <Hds::Button
+            @text='Follow the guide'
```
<img width="820" alt="image" src="https://user-images.githubusercontent.com/10027860/219813948-7820bbac-0016-4365-8d5b-afe0495ee625.png">
updated everywhere for zero-state and document links/external

valeriia-ruban commented on ui/packages/consul-ui/app/components/consul/acl/disabled/index.hbs:
```diff
@@ -15,11 +15,25 @@
-          <a href="{{env 'CONSUL_DOCS_URL'}}/acl/index.html" rel="noopener noreferrer" target="_blank">Read the documentation</a>
+        <li>
+          <Hds::Button
+            @text='Read the documentation'
```
done

valeriia-ruban commented:
> 
@jherschman , that's amazing review! so many good catches! ⚡ 
I've updated links from `hds::button` to `hds::link::standalone`

Reverted change for font-weight (it was [here](https://github.com/hashicorp/consul/pull/16251/files#diff-611e06d7dd761ed4852546f0beba0774627908df80f8cb31f846b763a4b41f07L70))
<img width="1035" alt="image" src="https://user-images.githubusercontent.com/10027860/219819758-65b845f4-4343-44f4-af69-4c681518d5a8.png">

as for the Exported Services - I hope it was overwritten by newer version of design-system. 
fixed addresses and exported services
<img width="820" alt="image" src="https://user-images.githubusercontent.com/10027860/219819705-dcb04747-a99a-41db-8223-10cf5b5c536a.png">

jherschman reviewed (approved):
@valeriia-ruban All changes are looking great, thank you!! 🔥  😄

WenInCode commented on ui/packages/consul-partitions/app/components/consul/partition/form/index.hbs:
```diff
@@ -103,7 +103,7 @@ as |State Guard ChartAction dispatch state|>
         type='reset'
         @color='secondary'
-        {{on 'click' (if @oncancel (fn @oncancel item) (fn @onsubmit item))}}
+        {{on 'click' (if @oncancel (fn (optional @oncancel item)) (fn (optional @onsubmit item)))}}
```
niiiice

WenInCode reviewed (approved):
LGTM 🚀
